tests/unit/dogstatsd/test_statsd.py

```diff
@@ -402,6 +403,22 @@ def test_accessing_socket_multiple_times_returns_same_socket(self):
         t.assert_equal(fresh_socket, dogpound.get_socket())
         t.assert_not_equal(FakeSocket(), dogpound.get_socket())
 
+    def test_tags_from_environment(self):
+        os.environ['DOGSTATSD_TAGS'] = 'country:china,age:45,blue'
+        statsd = DogStatsd()
+        del(os.environ['DOGSTATSD_TAGS'])
+        statsd.socket = FakeSocket()
+        statsd.gauge('gt', 123.4)
+        t.assert_equal('gt:123.4|g|#country:china,age:45,blue', statsd.socket.recv())
+
+    def test_tags_from_environment_and_constant(self):
+        os.environ['DOGSTATSD_TAGS'] = 'country:china,age:45,blue'
+        statsd = DogStatsd(constant_tags=['country:canada', 'red'])
+        del(os.environ['DOGSTATSD_TAGS'])
+        statsd.socket = FakeSocket()
+        statsd.gauge('gt', 123.4)
+        t.assert_equal('gt:123.4|g|#country:canada,red,country:china,age:45,blue', statsd.socket.recv())
+
```

Comment:
it's not super clear to me how to apply `preserve_environ_datadog` to these test methods cleanly. i went ahead and implemented the logic in a separate helper as a context manager.